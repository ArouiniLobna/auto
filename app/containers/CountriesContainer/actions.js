/*
 *
 * CountriesContainer actions
 *
 */

import * as types from './constants';

export function loadCountries() {
	return {
		type: types.LOAD_COUNTRIES
	};
}
export function loadCountriesSuccess(countries) {
	return {
		type: types.LOAD_COUNTRIES_SUCCESS,
		countries
	};
}

export function loadCountriesError(error) {
	return {
		type: types.LOAD_COUNTRIES_ERROR,
		error
	};
}
