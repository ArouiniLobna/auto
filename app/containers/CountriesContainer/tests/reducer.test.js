// import produce from 'immer';
import countriesContainerReducer from '../reducer';
import { loadCountries, loadCountriesError, loadCountriesSuccess } from '../actions';


/* eslint-disable default-case, no-param-reassign */
describe('countriesContainerReducer', () => {
  let state;
  beforeEach(() => {
    state = {
      countries: null,
  loadingCountries: null,
  countriesError: false,
    };
  });

  it('returns the initial state', () => {
    const expectedResult = state;
    expect(countriesContainerReducer(undefined, {})).toEqual(expectedResult);
  });

  it('handles the loadCountries action', () => {
    expect(countriesContainerReducer({}, loadCountries())).toMatchSnapshot();
  });

  it('handles the loadCountriesError action', () => {
    expect(countriesContainerReducer({}, loadCountriesError())).toMatchSnapshot();
  });

  it('handles the loadCountriesSuccess action', () => {
    expect(countriesContainerReducer({}, loadCountriesSuccess())).toMatchSnapshot();
  });

  /**
   * Example state change comparison
   *
   * it('should handle the someAction action correctly', () => {
   *   const expectedResult = produce(state, draft => {
   *     draft.loading = true;
   *     draft.error = false;
   *     draft.userData.nested = false;
   *   });
   *
   *   expect(appReducer(state, someAction())).toEqual(expectedResult);
   * });
   */
});
