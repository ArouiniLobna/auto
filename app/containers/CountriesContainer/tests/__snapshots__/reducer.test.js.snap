// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countriesContainerReducer handles the loadCountries action 1`] = `
Object {
  "countriesError": false,
  "loadingCountries": true,
}
`;

exports[`countriesContainerReducer handles the loadCountriesError action 1`] = `
Object {
  "countriesError": undefined,
  "loadingCountries": false,
}
`;

exports[`countriesContainerReducer handles the loadCountriesSuccess action 1`] = `
Object {
  "countries": undefined,
  "countriesError": false,
  "loadingCountries": false,
}
`;
